### PR TITLE
allow overriding sed executable in makefile

### PR DESCRIPTION
**Pull request by rolandwalker**
because some of the existing commands require GNU sed.  In particular, I needed to

```bash
make update-docs SED=gsed
```

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -74,6 +74,7 @@ DOCBOOK2PDF ?= docbook2pdf
 
 LCOV ?= lcov
 GENHTML ?= genhtml
+SED ?= sed
```
Comment (jonas) [inline Makefile]: Given that all scripts assume GNU sed, I think it is safer to default to `gsed`.

Comment (jonas) [inline Makefile]: OK, looks like this was a bad idea. I'll reenable my `sed -> gsed` alias and maybe configure a build that runs on macOS to ensure that the portability fixes you did in the test suite won't regress.

**Comment by rolandwalker** [comment]
Sec, it shouldn't be hard to check for both and prefer `gsed`, since GNU make is already needed for this Makefile.

**Comment by jonas** [comment]
Sweet!

**Pull request merged**